docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -129,13 +140,52 @@ For an end user, the UX to update the kubernetes version of the control plane an
 As an end user, I want to be able to easily scale up/down the number of replicas for each set of worker nodes in the cluster.
 - Currently, (for a cluster with 3 machine deployments) this is possible by updating these three different objects representing the sets of worker nodes in the pool. An easier user experience would be to update a single object to enable the scaling of multiple sets of worker nodes.
 
+#### Story 4 - Use ClusterClass to easily modify Clusters
+As an end user, I want to be able to easily change the configuration of all Clusters of a `ClusterClass`. For example, I want to be able to change the ControlPlane configuration by  
+changing the ControlPlane ref to point to another ControlPlane with a different configuration. The same should be possible for all templates referenced in a ClusterClass.
+
+**Notes**:
+- Only compatible changes (as specified in [ClusterClass compatibility](#clusterclass-compatibility)) should be allowed.
+- Changes to InfrastructureMachineTemplates and BootstrapTemplates should be rolled out according to the established operational practices documented in 
+[Updating Machine Infrastructure and Bootstrap Templates](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html), i.e. "template rotation".
+- There are provider-specific incompatible changes which cannot be validated in a "core" webhook, e.g. changing an immutable field of `KubeadmControlPlane`. Those changes will inevitably lead to
+  errors during topology reconciliation. Those errors should be surfaced on the Cluster resource.
+
+#### Story 5 - Ability to customize individual Clusters via variables
+As an end user, I want to customize individual Clusters simply by providing variables in the Cluster resource.
+
+**Note**: Without this ability all Clusters of the same ClusterClass would be the same apart from the properties that are already configure via the topology,
+like Kubernetes version, labels and annotations. This would limit the number of variants a single ClusterClass could address. Variables can e.g. be used to make 
+region/location, machine types and network configuration configurable.
+
+**Example**: A user wants to deploy CAPA Clusters using ClusterClass in different AWS regions. One option to achieve this is to duplicate the ClusterClass and its referenced templates.
+The better option is to introduce a variable and a corresponding patch in the ClusterClass. Now, a user can simply set the AWS region via a variable in the Cluster spec, instead of 
+having to duplicate the entire ClusterClass just to set a different region in the AWSCluster resource.
+
+#### Story 6 - Ability to mutate variables
+As an end user, I want to be able to modify variables in a Cluster. The variable modification should trigger a reconciliation which rolls out the managed topology accordingly, i.e. 
```

Comment:
Similar to https://github.com/kubernetes-sigs/cluster-api/pull/5212/files#r712039251
I feel this is more of a technical ability description that could may be conflated with story 4 or 5 to support a real world scenario which is the actual user story?

Reply:
I'll keep the stories separate (story 4 talks about general mutability without considering patches at all, story 5 already has a big scope). I'll add an example and drop the technical detail about patching